env0/resource_user_project_assignment.go

```diff
@@ -0,0 +1,116 @@
+package env0
+
+import (
+	"context"
+	"log"
+
+	"github.com/env0/terraform-provider-env0/client"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceUserProjectAssignment() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceUserProjectAssignmentCreate,
+		UpdateContext: resourceUserProjectAssignmentUpdate,
+		ReadContext:   resourceUserProjectAssignmentRead,
+		DeleteContext: resourceUserProjectAssignmentDelete,
+
+		Schema: map[string]*schema.Schema{
+			"user_id": {
+				Type:        schema.TypeString,
+				Description: `id of the user. Note: can also be an id of a "User" API key`,
+				Required:    true,
+				ForceNew:    true,
+			},
+			"project_id": {
+				Type:        schema.TypeString,
+				Description: "id of the project",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"role": {
+				Type:             schema.TypeString,
+				Description:      "the assigned role (Admin, Planner, Viewer, Deployer)",
+				Required:         true,
+				ValidateDiagFunc: ValidateRole,
```

Comment:
Doesn't the provider framework have a way of specifying some enum in a more native way than a validation func?

Reply:
@yaronya , maybe? I honestly don't know (couldn't find anything). In general, enum is not a valid schema type in the SDK. The AWS provider is doing something similar to us: https://github.com/hashicorp/terraform-provider-aws/issues/14601
If you know of another way please let me know and I will fix it in a future PR.